src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -885,6 +887,18 @@ public override void VisitObjectSyntax(ObjectSyntax syntax)
             {
                 var errors = new List<ErrorDiagnostic>();
 
+                var duplicatedProperties = syntax.Properties
+                    .GroupByExcludingNull(prop => prop.TryGetKeyText(), LanguageConstants.IdentifierComparer)
+                    .Where(group => group.Count() > 1);
+
+                foreach (var group in duplicatedProperties)
+                {
+                    foreach (ObjectPropertySyntax duplicatedProperty in group)
+                    {
+                        errors.Add(DiagnosticBuilder.ForPosition(duplicatedProperty.Key).PropertyMultipleDeclarations(group.Key));
+                    }
+                }
```

Comment:
Do we need a similar check where `ObjectTypeSyntax` is interpreted in [`DeclaredTypeManager`](https://github.com/Azure/bicep/blob/429182fadcd46ef3c1be550383d0589d561f7200/src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs#L518)?

Reply:
Good catch. It seems we don't have the validation for duplicate object type key today? I'll add a similar check and tests.